build(deps): update to angular 13

divStar opened the pull request:
This PR makes it so `ngx-pagination-data-source` works with Angular 13 projects (which is what I need).

I am not sure what else to do or run in order to have this version released somehow, but I'll gladly assist if need be :).

Thank you for making this!

Edit: migrating TSLint (deprecated) to ESLint

divStar commented:
This library should now support Angular 13 along with other new libraries.

Please have a look at the `.eslint.rc` settings:
```json
        "prefer-arrow/prefer-arrow-functions": "off",
        "semi": "off",
        "@typescript-eslint/semi": "off",
        "@typescript-eslint/member-delimiter-style": "off",
        "@typescript-eslint/member-ordering": "off",
        "id-blacklist": "off"
```

Since I was not sure what linting configuration you would prefer, I turned some rules off entirely. Please revisit this if you deem it important.

Tests seem to work fine. Sadly I am not sure how to check whether the library actually works in an Angular 13 application since I am not sure how to "locally publish" it.

nilsmehlhorn commented on projects/ngx-pagination-data-source/package.json:
```diff
@@ -1,6 +1,6 @@
 {
   "name": "ngx-pagination-data-source",
-  "version": "4.2.0",
+  "version": "4.2.1",
```
You don't need to bump the version here manually, the project follows semantic versioning based on conventional commits. We'll squash this PR into the master branch and run `npm run bump` and it'll infer the new version based on the incorporated changes - which in this case would rather entail a new major version. Note that multiple PRs can go into a single release, so bumping the version individually in each one manually gets hard :)

```suggestion
  "version": "4.2.0",
```

nilsmehlhorn commented on projects/ngx-pagination-data-source/package.json:
```diff
@@ -23,9 +23,9 @@
-    "@angular/common": "^12.0.0",
-    "@angular/core": "^12.0.0",
-    "@angular/cdk": "^12.0.0"
+    "@angular/common": "^13.1.1",
+    "@angular/core": "^13.1.1",
+    "@angular/cdk": "^13.1.1"
```
I'm not entirely sure on this, but I've always set this version to the lowest corresponding Angular major, so that anyone with Angular 13 can use the library without warnings. We build against the most recent version, but the library should be fine for consumption with any Angular 13 (maybe even lower versions, but that'd complicate things).

```suggestion
    "@angular/common": "^13.0.0",
    "@angular/core": "^13.0.0",
    "@angular/cdk": "^13.0.0"
```

nilsmehlhorn commented:
Thank you so much for the contribution @divStar! I've already taken a quick look and noted two things that caught my eye. I'll checkout the PR next chance I get to take a deeper look. 

I assume you've followed the steps at [update.angular.io](https://update.angular.io/)? Did that include the lint migration or did you run something else for that? :)

divStar commented:
I did try to, most of the relevant changes had been made automatically - except for the rules I added in the `.eslintrc.json` file. I also managed to install this version in an Angular 13 app I am developing in my spare time for fun, but sadly with mixed success.

In particular I keep having problems when setting up the dataSource and I don't know whether it's because of the changes I made - or because something changed due to the upgrade.

The errors I have seem kinda unrelated and not even in my code, but in Angular Material, but they didn't occur before I added this library. I'll investigate later.

I'll implement your change requests later today, too.

It really is a nice little library. I wish the example you made (especially that StackBlitz one on your website) had a proper backend call (perhaps intercepted by a mock service worker or something), because that had been one issue I had when following the otherwise awesome tutorial.

Anyway: thank you for your time and input. I'll update this PR.

divStar commented:
@nilsmehlhorn I reverted the version numbers (main one to `4.2.0` and that of the `peerDependencies` to `^13.0.0`).

divStar commented:
@nilsmehlhorn I know you're probably super busy (cause that's the world of a software developer really), but would you be able / willing to merge this PR?

Also note my comment in regards of not being able to use RXJS 7.+ (apparently Observer/Subject whatever has different attributes and/or behavior and thus it didn't work and I had to resort to RXJS 6.5.7 I think, because Angular 13 AND your PaginationDataSource both work with it).

nilsmehlhorn commented:
Sorry, yeah I've been busy and on holiday. I'll get this done now right away, might decide to roll my own PR though. Not meant to disregard your work in any way, but this way I can run the update with the latest version and update all my projects in the same way. :slightly_smiling_face:

divStar commented:
Hi there, I wouldn't have been upset if you decided to roll your own PR. I only care for the package to somehow be able to deal with Angular 13 really 🤣.

nilsmehlhorn commented:
I've just released a new major version to npm, thanks again for your effort! :slightly_smiling_face:
